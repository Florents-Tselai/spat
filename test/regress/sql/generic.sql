SET client_min_messages = ERROR;

SELECT SPAT_DB_NAME(); -- spat-default
-- switch to a new db
SET spat.db = 'test-spat';
SELECT SPAT_DB_NAME(); -- test-spat
SELECT SPAT_DB_CREATED_AT() NOTNULL;

-- switch back to the default to preserve my sanity when copy-pasting

SET spat.db = 'spat-default';

/* -------------------- STRINGS / GENERIC -------------------- */

-- SET - DELETE - SIZE
SELECT SPSET('tkey1', 'dfgf');
SELECT SPSET('tkeygdfg1', 'dfgf');
SELECT SPSET('gdf', 'dfgf');
SELECT SPSET('gfd', 'dfgf');

SELECT SP_DB_NITEMS(); --4

-- DELETE and inspect size
SELECT DEL('tkey1'); --existing
SELECT DEL('aaaa'); --not existing
SELECT SP_DB_NITEMS(); --3
SELECT DEL('tkeygdfg1');
SELECT DEL('gdf');
SELECT DEL('gfd');
SELECT SP_DB_NITEMS(); --0

-- SET / GET / DEL text
SELECT SPSET('key1', 'value1');
SELECT SPGET('key1');

SELECT DEL('key1');
SELECT SPGET('key1');

-- DB SIZE BYTES
SELECT SP_DB_SIZE_BYTES();
SELECT SP_DB_SIZE();

-- sptype
SELECT SPSET('key1', 'value1');
SELECT SPTYPE('key1');

SELECT SPTYPE('gsdgdf');
