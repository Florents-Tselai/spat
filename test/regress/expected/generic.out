SET client_min_messages = ERROR;
SELECT SPAT_DB_NAME(); -- spat-default
 spat_db_name 
--------------
 spat-default
(1 row)

-- switch to a new db
SET spat.db = 'test-spat';
SELECT SPAT_DB_NAME(); -- test-spat
 spat_db_name 
--------------
 test-spat
(1 row)

SELECT SPAT_DB_CREATED_AT() NOTNULL;
 ?column? 
----------
 t
(1 row)

-- switch back to the default to preserve my sanity when copy-pasting
SET spat.db = 'spat-default';
/* -------------------- STRINGS / GENERIC -------------------- */
-- SET - DELETE - SIZE
SELECT SPSET('tkey1', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT SPSET('tkeygdfg1', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT SPSET('gdf', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT SPSET('gfd', 'dfgf');
 spset 
-------
 dfgf
(1 row)

SELECT SP_DB_NITEMS(); --4
 sp_db_nitems 
--------------
            4
(1 row)

-- DELETE and inspect size
SELECT DEL('tkey1'); --existing
 del 
-----
 t
(1 row)

SELECT DEL('aaaa'); --not existing
 del 
-----
 f
(1 row)

SELECT SP_DB_NITEMS(); --3
 sp_db_nitems 
--------------
            3
(1 row)

SELECT DEL('tkeygdfg1');
 del 
-----
 t
(1 row)

SELECT DEL('gdf');
 del 
-----
 t
(1 row)

SELECT DEL('gfd');
 del 
-----
 t
(1 row)

SELECT SP_DB_NITEMS(); --0
 sp_db_nitems 
--------------
            0
(1 row)

-- SET / GET / DEL text
SELECT SPSET('key1', 'value1');
 spset  
--------
 value1
(1 row)

SELECT SPGET('key1');
 spget  
--------
 value1
(1 row)

SELECT DEL('key1');
 del 
-----
 t
(1 row)

SELECT SPGET('key1');
 spget 
-------
 
(1 row)

-- DB SIZE BYTES
SELECT SP_DB_SIZE_BYTES();
 sp_db_size_bytes 
------------------
          1048576
(1 row)

SELECT SP_DB_SIZE();
 sp_db_size 
------------
 1024 kB
(1 row)

-- sptype
SELECT SPSET('key1', 'value1');
 spset  
--------
 value1
(1 row)

SELECT SPTYPE('key1');
 sptype 
--------
 string
(1 row)

SELECT SPTYPE('gsdgdf');
 sptype 
--------
 null
(1 row)

